evm/src/keccak_sponge/keccak_sponge_stark.rs

```diff
@@ -477,6 +480,38 @@ impl<F: RichField + Extendable<D>, const D: usize> KeccakSpongeStark<F, D> {
         // indicating that it's a dummy/padding row.
         KeccakSpongeColumnsView::default().into()
     }
+
+    /// Expects input in *column*-major layout
+    fn generate_range_checks(&self, cols: &mut Vec<Vec<F>>) {
+        debug_assert!(cols.len() == NUM_KECCAK_SPONGE_COLUMNS);
+
+        let n_rows = cols[0].len();
+        debug_assert!(cols.iter().all(|col| col.len() == n_rows));
+
+        for i in 0..BYTE_RANGE_MAX {
+            cols[RANGE_COUNTER][i] = F::from_canonical_usize(i);
+        }
+        for i in BYTE_RANGE_MAX..n_rows {
+            cols[RANGE_COUNTER][i] = F::from_canonical_usize(BYTE_RANGE_MAX - 1);
+        }
```

Comment:
```suggestion
        let filler = F::from_canonical_u8(u8::MAX);
        for i in BYTE_RANGE_MAX..n_rows {
                    cols[RANGE_COUNTER][i] = filler;
        }
```

…to avoid creating a field element for every iteration? (Although I bet the compiler does that optimization anyway)

Reply:
Yeah, it's being used in a lot of places and I'd bet this is optimized away. One advantage to the current approach is that in the general case, it is clearly apparent which value we're writing to the trace / constraining, while saving in a temporary variable may not be clearly understandable.